test/ft/exceptions.cpp

```diff
@@ -231,4 +231,23 @@ test exception_and_unexpected_event = [] {
   sm.process_event(e1{});
   expect(sm.is(s1));
 };
+
+test exception_and_context = [] {
+  struct c {
+    auto operator()() const {
+      using namespace sml;
+      // clang-format off
+        return make_transition_table(
+           *idle + event<e1> / [] { throw exception1{}; } = s1
+          , s1 + exception<exception1> / [](const context &,const auto &){} = X
```

Comment:
Your unit-test seems not to compile.

It bails out with `Type T has to be passed via constructor!` error:

```
In file included from test/ft/exceptions.cpp:8:0:
include/boost/sml.hpp: In instantiation of ‘T& boost::sml::v1_1_0::aux::try_get(...) [with T = const context&]’:
include/boost/sml.hpp:375:34:   required from ‘boost::sml::v1_1_0::aux::pool<Ts>::pool(boost::sml::v1_1_0::aux::init&&, boost::sml::v1_1_0::aux::pool<TArgs ...>&&) [with TArgs = {context*}; Ts = {const context&}]’
include/boost/sml.hpp:1310:113:   required from ‘boost::sml::v1_1_0::back::sm< <template-parameter-1-1> >::sm(TDeps&& ...) [with TDeps = {context*}; typename boost::sml::v1_1_0::aux::enable_if<boost::sml::v1_1_0::aux::is_unique<boost::sml::v1_1_0::aux::none_type, TDeps ...>::value, int>::type <anonymous> = 0; TSM = boost::sml::v1_1_0::back::sm_policy<<lambda()>::c>]’
include/boost/sml.hpp:1715:80:   required from here
include/boost/sml.hpp:333:3: error: static assertion failed: Type T has to be passed via constructor!
   static_assert(never<T>::value, "Type T has to be passed via constructor!");
   ^~~~~~~~~~~~~
include/boost/sml.hpp:334:1: error: no return statement in function returning non-void [-Werror=return-type]
 }
 ^
```
Was that the error (message) you tried to fix?
Or is this error appearing for the same reason as described in #139 and #140?

Reply:
I'm pretty sure that compiled with g++ 7.2, which compiler are you testing on? I'll change the test to use bool instead of context so that is the same as the other tests.

Reply:
I tried with GCC (versions 7.2, 6.3 and 5.4) and Clang (version 3.8). All fail.

The build-bot (travis) seams to have the same problem.

Reply:
At a cursory glance, that indeed seems to be the same issue as #139 and #140. I've added a test specific to the issue in pull request #140, so if all goes well this issue should be fixed soon.

Reply:
@Bagira80 @AnthonyVH  good catch, it looks like bug #140. I've changed the test to take the context object by non-const reference.